### PR TITLE
Allocate slots for each possible huge blob size

### PR DESCRIPTION
**Pull request by qjex**

### CODE REVIEW

File: ya.conf [hunk 1]
```diff
@@ -33,3 +33,4 @@ OPENSOURCE = "yes"
 USE_PREBUILT_TOOLS = "no"
 APPLE_SDK_LOCAL = "yes"
 USE_CLANG_CL = "yes"
+BUILD_LANGUAGES="CPP PY3 PY2 GO"
```
Comment (alexvru) [inline ya.conf:36]: Do we really need it here in this review?

File: ydb/core/blobstorage/vdisk/common/vdisk_hugeblobctx.h [hunk 2]
```diff
@@ -71,11 +71,13 @@ namespace NKikimr {
         // check whether this NEW blob is huge one; userPartSize doesn't include any metadata stored along with blob
         bool IsHugeBlob(TBlobStorageGroupType gtype, const TLogoBlobID& fullId) const;
 
-        THugeBlobCtx(ui32 minREALHugeBlobInBytes, const std::shared_ptr<const THugeSlotsMap> &hugeSlotsMap, bool addHeader)
-            : MinREALHugeBlobInBytes(minREALHugeBlobInBytes)
+        THugeBlobCtx(ui32 minHugeBlobInBytes, ui32 appendBlockSize, const std::shared_ptr<const THugeSlotsMap> &hugeSlotsMap, bool addHeader)
+            : MinREALHugeBlobInBytes(minHugeBlobInBytes / appendBlockSize * appendBlockSize + 1)
```
Comment (alexvru) [inline ydb/core/blobstorage/vdisk/common/vdisk_hugeblobctx.h:75]: + 1?

File: ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp [hunk 3]
```diff
@@ -408,32 +408,27 @@ namespace NKikimr {
...
+                ui32 minBlobInBytes,
                 ui32 milestoneBlobInBytes,
                 ui32 maxBlobInBytes,
                 ui32 overhead,
                 bool oldMapCompatible)
             : VDiskLogPrefix(vdiskLogPrefix)
             , ChunkSize(chunkSize)
             , AppendBlockSize(appendBlockSize)
-            , MinHugeBlobInBytes(minHugeBlobInBytes)
+            , MinBlobInBytes(minBlobInBytes)
```
Comment (alexvru) [inline ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp:419]: What's the point of removing "Huge" infix from the identifier?

File: ydb/core/blobstorage/vdisk/localrecovery/localrecovery_public.cpp [hunk 4]
```diff
@@ -398,7 +398,7 @@ namespace NKikimr {
             Y_UNUSED(ctx);
             const ui32 blocksInChunk = LocRecCtx->PDiskCtx->Dsk->ChunkSize / LocRecCtx->PDiskCtx->Dsk->AppendBlockSize;
             Y_ABORT_UNLESS(LocRecCtx->PDiskCtx->Dsk->AppendBlockSize * blocksInChunk == LocRecCtx->PDiskCtx->Dsk->ChunkSize);
-
+
```
Comment (alexvru) [inline ydb/core/blobstorage/vdisk/localrecovery/localrecovery_public.cpp:401]: Dangling whitespace.

File: ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp (hunk 3 above)
Comment (qjex) [reply ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp:419]: I think it's more consistent with the left boundary naming. Now the range is presented by [MinBlobInBytes, MaxBlobInBytes].

File: ydb/core/blobstorage/vdisk/common/vdisk_hugeblobctx.h (hunk 2 above)
Comment (qjex) [reply ydb/core/blobstorage/vdisk/common/vdisk_hugeblobctx.h:75]: Yes, this is the way it works now. So, I am just preserving the current behaviour here.  
Please, see the TAllChains::GetMinREALHugeBlobInBytes

File: ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp (hunk 3 above)
Comment (alexvru) [reply ydb/core/blobstorage/vdisk/huge/blobstorage_hullhugeheap.cpp:419]: But the "min blob size" is 1 byte. And the "min huge blob size" is absolutely different metric.
It looks like
1b <-- small blobs --> minHugeBlobSize <-- huge blobs --> maxBlobSize

**Review by alexvru — APPROVED** [review]

**Pull request merged**